code/modules/cm_aliens/hivebuffs/hivebuff.dm

```diff
@@ -336,38 +331,54 @@
 	hive.stored_larva += 5
 	return TRUE
 
-/datum/hivebuff/extra_life
-	name = "Boon of Plenty"
-	desc = "Increases all xenomorph health by 20 for 5 minutes"
+/datum/hivebuff/evo_buff
+	name = "Boon of Evolution"
+	desc = "Doubles evolution speed for 5 minutes."
 	tier = HIVEBUFF_TIER_MINOR
-	engage_flavourmessage = "The Queen has imbued us with greater fortitude."
+	engage_flavourmessage = "The Queen has blessed us with faster evolution."
 	duration = 5 MINUTES
 	number_of_required_pylons = 1
+	var/value_before_buff
 
-/datum/hivebuff/extra_life/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
-	xeno.health_modifier += XENO_HEALTH_MOD_VERY_SMALL
-	xeno.recalculate_health()
+/datum/hivebuff/evo_buff/on_engage(obj/effect/alien/resin/special/pylon/purchased_pylon)
+	var/value_before_buff = SSxevolution.get_evolution_boost_power(hive.hivenumber)
+	hive.override_evilution(value_before_buff * 2, TRUE)
 
-/datum/hivebuff/extra_life/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
-	xeno.health_modifier -= XENO_HEALTH_MOD_VERY_SMALL
-	xeno.recalculate_health()
+	return TRUE
 
-/datum/hivebuff/extra_life/major
-	name = "Major Boon of Plenty"
-	desc = "Increases all xenomorph health by 40 for 10 minutes"
+/datum/hivebuff/evo_buff/on_cease()
+	if(SSxevolution.get_evolution_boost_power(hive.hivenumber) == XENO_HIJACK_EVILUTION_BUFF)
```

Comment:
This assumes that the value we set would never be set to XENO_HIJACK_EVILUTION_BUFF. It would be better to check if say `/datum/dropship_hijack/almayer/proc/fire()` sets a var true or something on fire. SStimer.mode only sets `is_in_endgame` when the shuttle crashes.

Also xenos will be restored to the buffed amount if this buff was on during hijack. So you need to add a signal or check if overriden or something prior to `/obj/structure/machinery/computer/shuttle/dropship/flight/proc/hijack` reading the "original" evolution value.

Reply:
It wouldn't be a bad thing to just have a signal for `/obj/structure/machinery/computer/shuttle/dropship/flight/proc/hijack` or for that proc to just tell the hive to cease all active buffs on hijack.